Issue #485: Fix linting

nikosbosse opened the pull request:
<!-- Thanks for opening this pull request! Below we have provided a suggested template for PRs to this repository and a checklist to complete before opening a PR -->
 
## Description

This PR closes #485

This PR fixes linting issues on the current development branch. It
- adds R files based on the vignettes to the files ignored by the linter (these are automatically generated based on the vignettes and always contain linting issues)
- fixes various linting issues across a large number of files. 

Two single items stand out: 
1. the function name `interval_coverage_deviation_quantile` is too long
2. the function name `plot.scoringutils_available_forecasts` is too long

We could simply change `interval_coverage_deviation_quantile` to `interval_coverage_dev_quantile` (exactly 30 chars long). An alternative would be to simply exclude the line from linting. 
The second one is related to #403. 

_**Update 1** I created a second PR for the name of `plot.scoringutils_available_forecasts()`, #511, which addresses #403_ 
_**Update 2** I renamed the function to `interval_coverage_dev_quantile`_ 

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title as follows: *issue-number*: PR title
- [x] I have tested my changes locally.
- [x] ~~I have added or updated unit tests where necessary.~~
- [x] ~~I have updated the documentation if required.~~
- [x] I have built the package locally and run rebuilt docs using roxygen2.
- [x] My code follows the established coding standards and I have run `lintr::lint_package()` to check for style issues introduced by my changes. 
- [x] ~~I have added a news item linked to this PR.~~
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

<!-- Thanks again for this PR - @scoringutils dev team -->

seabbs reviewed (approved):
Definitely looks better. I think shortening the name is fine. Still failing as its flagging all the lines that are new and don't have test coverage.

nikosbosse commented on .lintr:
```diff
@@ -11,6 +11,6 @@ linters: linters_with_tags(
     list.files("tests", recursive = TRUE, full.names = TRUE),
     list.files("inst", recursive = TRUE, full.names = TRUE),
-    "vignettes/metric-details.Rmd"
+    list.files("vignettes", pattern = ".R$", full.names = TRUE)
```
I don't know 👀 But apparently we always get an R version extracted from the vignettes....

nikosbosse replied:
Or is "these" about all the exceptions? Happy to reconsider them.

seabbs replied:
No I mean why do you have these R scripts? When are they being created/by what process? I don't think we need or want them.

nikosbosse commented on R/add_coverage.R:
```diff
@@ -51,13 +51,9 @@ add_coverage <- function(data) {
-    TRUE,
-    FALSE)
+  interval_data[
+    , interval_coverage := (observed <= upper) & (observed >= lower)
```
I honestly have no idea how to correctly reformat `data.table` code...
Is this better? 
<img width="578" alt="image" src="https://github.com/epiforecasts/scoringutils/assets/37978797/a4e5e9ee-cff1-4c0a-a942-8c42be976f5c">

seabbs replied:
dt[, 
	stuff
]

seabbs replied:
only one tab for stuff not in line with the ,

nikosbosse replied:
Implemented that. It is definitely not what the automatic styling suggests (i.e. when you press Enter to create a new line you end up with what I showed in my screenshot). But the linter also doesn't complain about your code, so I went with that and will try to adhere to that style in the future.

seabbs commented on R/data.R:
```diff
@@ -215,13 +230,15 @@
+#' - "coverage_50" = [interval_coverage_quantile()]
+#' - "coverage_90" = \(...) \{
+#'      run_safely(..., range = 90, fun = [interval_coverage_quantile])
+#'   \}
```
as in I thought it was () brackets

nikosbosse replied:
Apparently ( ) brackets works as well (but then embracing everything). If you don't have a strong preference I'll leave as is for now

nikosbosse commented on R/utils_data_handling.R:
```diff
@@ -65,12 +65,14 @@ merge_pred_and_obs <- function(forecasts, observations,
 
   # see whether the column name as well as the content is the same
-  overlapping <- (as.list(combined[, ..colnames_x]) %in% as.list(combined[, ..colnames_y])) & basenames_x == basenames_y
+  overlapping <- (
```
I refactored it a bit to make it more legible

nikosbosse commented:
I think I addressed all your comments. Are you happy for me to merge? In particular, should we address tests in a different PR, or would you prefer doing this now?

seabbs reviewed (approved):
LGTM
